app/db/queries/application.py

```diff
@@ -545,3 +610,136 @@ def _delete_all_components_in_pages(page_ids):
     stmt = delete(Component).filter(Component.page_id.in_(page_ids))
     db.session.execute(stmt)
     db.session.commit()
+
+
+# Section and form reordering
+
+
+def move_section_down(round_id, section_index_to_move_down: int):
+    """Moves a section one place down in the ordered list of sections in a round.
+    In this case down means visually down, so the index number will increase by 1.
+
+    Element | Section.index | Index in list
+        A   |   1           |   0
+        B   |   2           |   1
+        C   |   3           |   2
+
+    Then move B down, which results in C moving up
+
+    Element | Section.index | Index in list
+        A   |   1           |   0
+        C   |   2           |   1
+        B   |   3           |   2
+
+    Args:
+        round_id (UUID): Round ID to move this section within
+        section_index_to_move_down (int): Current Section.index value of the section to move
+    """
+    round: Round = get_round_by_id(round_id)
+    if section_index_to_move_down >= len(round.sections):
+        raise IndexError("Cannot move the last section down")
+    list_index_to_move_down = section_index_to_move_down - 1  # Need the 0-based index inside the list
+    # Remove the seciton we are moving from the list
+    moved_section = round.sections.pop(list_index_to_move_down)
+    # Reinsert it at the new position - because the list index is 0-based, this equates to the old Section.index
+    round.sections.insert(section_index_to_move_down, moved_section)
```

Comment:
Am I right in thinking this is just an element swap? If so, can use bit of Python syntactic sugar :
```
round.sections[list_index_to_move_down],  round.sections[list_index_to_move_down+1] = round.sections[list_index_to_move_down+1], round.sections[list_index_to_move_down]
```

Reply:
Wow... mind blown... 🤯 Thank you, I will have a play with this!